### ISSUE TITLE
Add Tweedie distribution to torch.distributions

### ISSUE THREAD

**Issue by StefanoDamato** [pytorch/pytorch#172059]
### 🚀 The feature, motivation and pitch

I propose to include the Tweedie distribution to `torch.distributions`. 
A draft implementation is available with tests in PR #171705.

**Description**

The Tweedie family (Tweedie, 1984) is a class of exponential dispersion models (a generalisation of the exponential family), parameterised by the mean $\mu$, the dispersion $\phi$ and the power $\rho$, such that its variance is equal to  \phi \mu^\rho$$ (Jørgensen, 1987).

**Evaluation**

For $\rho \in (1,2)$, the distribution can be characterised as a compound Poisson-Gamma representation. The evaluation of its density function is non-trivial, as it requires to truncate an infinite sum (Dunn and Smyth, 2005). The sampling easily leverages the Poisson-Gamma structure.

**Use cases**

The Tweedie distribution is used to model zero-inflated data with possibly large real-valued spikes. This is often observed, for instance, in insurance (Yang et al., 2018). An approximation has also been used to predict intermittent demand (Jeong and Seong, 2022).

**References**
- Tweedie, M. C. K. (1984). *An index which distinguishes between some important exponential families*. **Statistics: Applications and New Directions** (pp. 579–604). Indian Statistical Institute.
- Jørgensen, B. (1987). *Exponential dispersion models*. **Journal of the Royal Statistical Society: Series B (Methodological)**, 49(2), 127–162.
- Dunn, P. K., & Smyth, G. K. (2005). *Series evaluation of Tweedie exponential dispersion model densities*. **Statistics and Computing**, 15(4), 267–280.
- Yang, X., Qian, W., & Zou, H. (2018). *Insurance premium prediction via gradient tree-boosted Tweedie compound Poisson models*. **Journal of Computational and Graphical Statistics**, 27(4), 841–854.
- Jeong, Y., & Seong, S. (2022). *Robust recurrent network model for intermittent time-series forecasting*. **International Journal of Forecasting**, 38(4), 1366–1381.

**Implementations**
- `tweedie` R package implements the distribution.
- `statmod` R package supports Tweedie GLMs.
- `tweedie` Python package implements the distribution.
- `statsmodels` Python package supports Tweedie GLMs.

### Alternatives

Ignore the PR

### Additional context

I used my `torch` implementation and I found it had very good performances when coupled with Gaussian Processes (implemented in `GPyTorch`). I expect it to have good results with neural networks too, and I hope it can be beneficial for the community..

cc @fritzo @neerajprad @alicanb @nikitaved
